### PR TITLE
#339 Make the nimforum Nim2 compatible

### PR DESCRIPTION
**Pull request by PhilippMDoerner**
The PR corresponding to issue #339, which should close it.

The changes provided in here were what was needed to get the tests (as executed via `nimble test`) to run locally while using the following nim version:

```txt
Nim Compiler Version 1.9.1 [Linux: amd64]
Compiled at 2022-12-22
Copyright (c) 2006-2022 by Andreas Rumpf

git hash: d0721eadf8cd6bf9436b5e5c9113c4c7bcfcc770
active boot switches: -d:release
```

The changes at large were mostly:

- Circumvent gc-safety checks of the compiler (I assume the checks have gotten stricter?) using cast(gcsafe)
- Move back to use refc instead of ORC in the nimble tasks. Using ORC leads to hard to debug error messages from within the C code that I don't really know how to deal with. As it worked previously fine under refc, I moved back to that. 
- Adjust Imports of smtp and db_sqlite since they are no longer part of the std lib
- Adding "quit()" to the `nimble testdb` task to **force** the task to return with exitCode 0. Previously it returned... some sort of number depending on the amount of text echo'd out. That lead to the test-task being unable to get past it as it interpreted that exitCode (as per the default specs of `exec`)  to be an error

### CODE REVIEW

**Comment by PhilippMDoerner** [comment]
The error appears to be:
```txt
stack trace: (most recent call last)
/tmp/nimblecache-946033505/nimscriptapi_865193738.nim(187, 16)
     Error: Exception raised during nimble script execution
/home/runner/work/nimforum/nimforum/nimforum.nimble(46, 10) frontendTask
/home/runner/.choosenim/toolchains/nim-1.6.10/lib/system/nimscript.nim(199, 19) cpFile
/home/runner/.choosenim/toolchains/nim-1.6.10/lib/system/nimscript.nim(199, 19) Error: unhandled exception: No such file or directory
Additional info: src/frontend/forum.js [OSError]
geckodriver: Terminated
/home/runner/work/nimforum/nimforum/tests/browsertester.nim(78) browsertester
/home/runner/work/nimforum/nimforum/tests/browsertester.nim(58) main
/home/runner/.choosenim/toolchains/nim-1.6.10/lib/system/assertions.nim(38) failedAssertImpl
/home/runner/.choosenim/toolchains/nim-1.6.10/lib/system/assertions.nim(28) raiseAssert
/home/runner/.choosenim/toolchains/nim-1.6.10/lib/system/fatal.nim(54) sysFatal
Error: unhandled exception: /home/runner/work/nimforum/nimforum/tests/browsertester.nim(58, 11) `execCmd("nimble -y frontend") == QuitSuccess`  [AssertionDefect]
Error: execution of an external program failed: '/home/runner/work/nimforum/nimforum/tests/browsertester '
       Tip: 33 messages have been suppressed, use --verbose to show them.
```

I'm not sure what that error is about.
Basically `nimble test` is failing, because it runs `browserTester.nim` which executes `nimble frontend`, which fails on this line:
`cpFile "src/frontend/forum.js", "public/js/forum.js"`

with this exception:
` Error: unhandled exception: No such file or directory`

I'm not sure what the problem here is or how I could fix it, any help?

**Comment by ringabout** [comment]
CI is now at 1.6.10, you can switch to/add 2.0 matrix.

**Comment by PhilippMDoerner** [comment]
By matrix you mean adding a pipeline that compiles under 2.0?

**Comment by ringabout** [comment]
yeah https://github.com/nim-lang/nimforum/blob/3eadc555e91d4390b3715b92e6d72ae3bd4c1864/.github/workflows/main.yml#L21

File: src/setup_nimforum.nim [hunk 1]
```diff
@@ -381,4 +386,4 @@ when isMainModule:
   else:
     echoHelp()
 
-
+  quit()
```
Comment (ringabout) [inline src/setup_nimforum.nim:389]: Why?

Comment (PhilippMDoerner) [reply src/setup_nimforum.nim:389]: > Adding "quit()" to the nimble testdb task to force the task to return with exitCode 0. Previously it returned... some sort of number depending on the amount of text echo'd out. That lead to the test-task being unable to get past it as it interpreted that exitCode (as per the default specs of exec) to be an error

File: nimforum.nimble [hunk 2]
```diff
@@ -24,21 +24,24 @@ requires "karax#45bac6b"
...
   exec "./src/forum"
 
 task testbackend, "Runs the forum backend in test mode":
-  exec "nimble c -r -d:skipRateLimitCheck src/forum.nim"
+  exec "nimble c -r --mm:refc -d:skipRateLimitCheck src/forum.nim"
 
 task frontend, "Builds the necessary JS frontend (with CSS)":
-  exec "nimble c -r src/buildcss"
-  exec "nimble js -d:release src/frontend/forum.nim"
+  exec "nimble c -r --mm:refc src/buildcss"
```
Comment (ringabout) [inline nimforum.nimble:44]: omission?

Comment (ringabout) [inline nimforum.nimble:44]:
```suggestion
  exec "nimble c -r --mm:refc src/buildcss"
  exec "nimble js -d:release src/frontend/forum.nim"
```

Comment (PhilippMDoerner) [reply nimforum.nimble:44]: Done

**Comment by PhilippMDoerner** [comment]
I forgot about an issue I quickly fixed locally but does mean the webdriver package needs an update:
Nim 2.0 forbids comparing strings directly to enums (at least there are no direct `==` proc for it around.
Webdriver makes use of them though. That leads to compiler issue. That means the webdriver package needs an update.

**Comment by PhilippMDoerner** [comment]
I have created a Pullrequest that should fix the issue, but we'll either have to fork webdriver or have to wait for dom to merge the PR:
https://github.com/dom96/webdriver/pull/14

**Pull request merged**